packages/modules/navigation/src/lib/MemoryStack.ts

```diff
@@ -0,0 +1,106 @@
+import type { HistoryStack, Location, LocationState, To } from './types';
+import { pathToString, resolvePath } from './utils';
+
+/**
+ * Memory-based history stack implementation.
+ *
+ * Stores navigation state in memory instead of using browser APIs.
+ * Unlike BrowserHistoryStack, push/replace only update the current location
+ * (history entries are managed by the reducer, not the stack).
+ */
+export class MemoryHistoryStack implements HistoryStack {
```

Comment:
I don't feel like this is a stack since it doesn't have a stack. Only `go` method takes state as arg and uses that as the stack.